### PR TITLE
Replace explicit Sentry capture with logger call

### PR DESCRIPTION
**Pull request by jhchabran**
Part of https://github.com/sourcegraph/sourcegraph/issues/35846

Because we have Sentry sinks, the error will still be reported, on
environment where a sentry DSN is given in the site-config.

## Test plan

Ran tests locally, really small change.

### CODE REVIEW

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/login.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ func CallbackHandler(config *oauth2.Config, success, failure http.Handler) http.
 }
 
 func gitlabHandler(config *oauth2.Config, success, failure http.Handler) http.Handler {
+	logger := log.Scoped("gitlab-handler", "Gitlab Handler")
```
Comment (bobheadxi) [inline enterprise/cmd/frontend/internal/auth/gitlaboauth/login.go:27]:
```suggestion
	logger := log.Scoped("gitlabHandler", "GitLab oAuth handler")
```

> Scope names should be CamelCase or camelCase, and the scope description should follow [the same conventions as a log message](https://docs.sourcegraph.com/dev/how-to/add_logging#writing-log-messages).

https://docs.sourcegraph.com/dev/how-to/add_logging#scoped-loggers

**Review by bobheadxi — COMMENTED** [review]
We might want to merge the log library update too: https://github.com/sourcegraph/sourcegraph/pull/37138

**Review by bobheadxi — APPROVED** [review]
oops meant to approve!

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/login.go (hunk 1 above)
Comment (unknwon) [reply enterprise/cmd/frontend/internal/auth/gitlaboauth/login.go:27]: "oAuth" should be "OAuth" :)

**Pull request merged**